zerver/views/auth.py

```diff
@@ -455,8 +455,11 @@ def start_social_login(request: HttpRequest, backend: str, extra_arg: Optional[s
             return redirect_to_config_error("saml")
         extra_url_params = {'idp': extra_arg}
 
+        if backend == "apple-id":
+            backend = "apple_id"
```

Comment:
Why do we need this conversion?

Reply:
It would be best if we can avoid this kind of thing

Reply:
It's because we are doing `backend.upper()` below to get `key_setting`. 
should I change it and add a new line with `if backend == "apple-id"`?

Reply:
How about we make the backend name just `"apple"`, and the setting names just `SOCIAL_AUTH_APPLE_TEAM` and so on?

That would resolve the need for any wrinkle in this logic.

It'd also make the settings names rather easier to understand, I think -- it's kind of confusing that e.g. the setting for the "Team ID" has a name that ends in `ID_TEAM`. Which is happening because of this collision, or coincidence, that the word "ID" is being used for two different things there in quick succession.

I see that upstream sets `name = "apple-id"`, but we're subclassing it and overriding that anyway already.

Reply:
Alternatively, "appleid". That's what Apple uses in the URLs that appear in this system -- like https://appleid.apple.com/auth/authorize and https://appleid.cdn-apple.com/appleid/button . And it avoids the confusion of reading e.g. "APPLE_ID_TEAM" like "Apple / ID team", as well as any need to juggle hyphens vs. underscores.